src/couch_httpd_auth.erl

```diff
@@ -346,10 +346,10 @@ handle_session_req(#httpd{method='GET', user_ctx=UserCtx}=Req, _AuthModule) ->
                 ]}},
                 {info, {[
                     {authentication_db, ?l2b(config:get("couch_httpd_auth", "authentication_db"))},
-                    {authentication_handlers, [auth_name(H) || H <- couch_httpd:make_fun_spec_strs(
-                            config:get("httpd", "authentication_handlers"))]}
```

Comment:
Well, quite naive one: move these to own micro-plugins (which could be still live in this repo) and provide config options to enable/disable these handlers directly.

As for filling UserCtx fields, may be just backport that logic from httpd? Sounds not harmful for the first sight.